services/storage/db.go

```diff
@@ -0,0 +1,99 @@
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const file = "prompt.db"
+const table = "prompts"
+
+type Prompt struct {
+	ID        string
+	Prompt    string
+	Response  string
+	CreatedAt time.Time
+}
+
+func openDB() (db *sql.DB) {
+	db, err := sql.Open("sqlite3", file)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return db
+}
+
+func execute_sql(sql string, args ...interface{}) (rows sql.Result, err error) {
+	db := openDB()
+	defer db.Close()
+
+	res, err := db.Exec(sql, args...)
+	if err != nil {
+		err = fmt.Errorf("execute_sql: %s .. %s", err, sql)
+		log.Fatal(err)
+	}
+	return res, err
+}
+
+func query_db(sql string) (rows *sql.Rows) {
+	db := openDB()
+	defer db.Close()
+
+	rows, err := db.Query(sql)
+	if err != nil {
+		err = fmt.Errorf("query_db: %s", err)
+		log.Fatal(err)
+	}
+	return rows
+}
+
+func CreateTable() {
+	tableSql := fmt.Sprintf("SELECT name FROM sqlite_master WHERE type='table' AND name='%s'", table)
+	rows := query_db(tableSql)
+	defer rows.Close()
+
+	exists := rows.Next()
+	if exists {
+		return
+	}
+
+	createSql := fmt.Sprintf("CREATE TABLE %s (id INTEGER PRIMARY KEY AUTOINCREMENT, prompt TEXT, response TEXT, CreatedAt DATETIME DEFAULT CURRENT_TIMESTAMP)", table)
```

Comment:
Alternatively, looks like sqlite supports `IF NOT EXISTS`

```suggestion
        createSql := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (id INTEGER PRIMARY KEY AUTOINCREMENT, prompt TEXT, response TEXT, CreatedAt DATETIME DEFAULT CURRENT_TIMESTAMP)", table)
```
[Found this in a blog](https://www.sqlitetutorial.net/sqlite-create-table/).